src/request/components/request-detail-panel-execution.jsx

```diff
@@ -336,6 +338,60 @@ var ParamaterList = React.createClass({
     }
 });
 
+var MiddlewareList = React.createClass({
+    render: function() {
+        var middlewareStartMessages = this.props.middlewareStartMessages;
+        var middlewareEndMessages = this.props.middlewareEndMessages;
+        var beforeExecuteCommandMessages = this.props.beforeExecuteCommandMessages;
+        var afterExecuteCommandMessages = this.props.afterExecuteCommandMessages;
+        var mongoDBMessages = this.props.mongoDBMessages;
+        
+        var content = null;
+        
+        if (middlewareStartMessages
+            && middlewareEndMessages
+            && middlewareStartMessages.length >= 0
+            && middlewareEndMessages.length >= 0) {
+        
+            middlewareStartMessages = middlewareStartMessages.sort(function (a,b) { return a.ordinal - b.ordinal; });
+            middlewareEndMessages = _.indexBy(middlewareEndMessages, 'payload.correlationId');
+            
+            var middlewareComponent = _.map(middlewareStartMessages, function (middlewareStartMessage) {
+                var middlewareEndMessage = middlewareEndMessages[middlewareStartMessage.payload.correlationId];
+                var middlewareEndPayload = middlewareEndMessage.payload;
+                
+                var result = middlewareEndPayload.result;
+                
+                switch (result) {
+                    case 'next': result = 'Next'; break;
+                    case 'end': result = 'End'; break;
+                    case 'error': result = 'Error'; break;
+                }
+                
+                return (
+                        <div className="tab-section-boxing">
+                            <section className="flex flex-row flex-inherit flex-base tab-section-item">
+                                <div className="tab-execution-important col-8">{middlewareEndPayload.name} &nbsp; <span className="text-minor">({result})</span></div>
```

Comment:
I'll remove it.  That said, it's still unclear to me whether the action/view has any real relevancy in the Node world.  The "action" is really just the last piece of executed middleware; it doesn't seem to have any other real significance.  View, yes probably, for those applications doing server-side, template-based rendering.  However, according to @nebrius, that is actually the minority compared to applications doing client-side rendering.